src/common/components/hosts/MassChangeHostnameModal.tsx

```diff
@@ -114,6 +117,16 @@ const withTemplate =
         ),
       };
     }
+
+    if (newHostnames.some((hostname) => FORBIDDEN_HOSTNAMES.includes(hostname || ''))) {
```

Comment:
I'd like to avoid the code duplication from lines 109 to 128.

Maybe we could extract the function that updates the `validationResult` as all it needs to know is the type of error. Or some other way, but the pattern is the same for the 2 cases.

Reply:
Also, I can enter `localhost{{n}}` and it tells me it's forbidden, but if I enter `localhost.localdomain{{n}}` it won't.
Shouldn't either both be accepted or rejected?

Reply:
`localhost.localdomain{{n}}` isn't forbidden. We only forbid hostnames that resolve to loopback addresses (see your `/etc/hosts` file for that. If you put in `localhost{{n}}.localdomain{{n}}`, that would be rejected.

Reply:
Oh, then what happens is that `localhost{{n}}` is rejected because it generates `localhost4` which is forbidden.
Thanks for the explanation!